extensions/likes/src/Listener/SendNotificationWhenPostIsLiked.php

```diff
@@ -22,7 +22,7 @@ public function __construct(
 
     public function handle(PostWasLiked $event): void
     {
-        if ($event->post->user && $event->post->user->id != $event->user->id) {
+        if (gettype($event->post->content) == 'string' && $event->post->user && $event->post->user->id != $event->user->id) {
```

Comment:
- It would be better to do an exact `===` comparison while using comparisons like these
- Instead of using `gettype`, just using the php `is_string()` method would probably be sufficient, it also reads better instead of seeing a `'string'` for something that is a native php type

My biggest gripe with this is however that there's no guarantee whatsoever that `content` is going to be just post body; what I would probably do is:

```suggestion
        if ($event->post->content instanceof CommentPost && $event->post->user && $event->post->user->id != $event->user->id) {
```

Now this gripe is based on the idea that notifications for likes would only be sent for posts that have readable content, but it would probably be better to completely separate the logic into a "Post Types that send Notifications" logic, so that it becomes flexible enough for extending; what do you think @flarum/core ?